project/iterative_sorting.py

```diff
@@ -2,29 +2,51 @@
 def selection_sort( arr ):
     # loop through n-1 elements
     for i in range(0, len(arr) - 1):
-        cur_index = i
-        smallest_index = cur_index
-
-        # TO-DO: find next smallest element
-        # (hint, can do in 3 loc) 
-        for j in range(cur_index, len(arr)):
-            if arr[j] < arr[smallest_index]:
-                smallest_index = j
+        current_index = i
+        smallest_index = current_index
+        # print(arr[i])
+        # print(arr)
+
+        # Find Smallest
```

Comment:
I love the comment trail here very easy to follow. Not mush else to say on these sorting methods other than that tbh. But writing out both the steps and the expected output is just great when you have to time to leave such a detailed trail of information. As always your commitment to learning is superb!